lib/ecto/schema.ex

```diff
@@ -1880,15 +1880,15 @@ defmodule Ecto.Schema do
     Module.put_attribute(mod, :changeset_fields, {name, {:assoc, struct}})
   end
 
-  @valid_embeds_one_options [:strategy, :on_replace, :source]
+  @valid_embeds_one_options [:strategy, :on_replace, :source, :on_cast]
```

Comment:
I would prefer to not introduce this option. `cast_embed` by itself is not going away... we are just allowing it to be handled on regular cast too.